app/views/handlers/question.py

```diff
@@ -226,12 +226,13 @@ def get_return_to_hub_url(self):
 
     def capture_dependent_sections_for_list(self, list_name):
         section_ids = self._schema.get_section_ids_dependent_on_list(list_name)
-        section_ids.append(self.current_location.section_id)
+        section_ids.add(self.current_location.section_id)
 
-        section_keys_to_add = self.questionnaire_store_updater.started_section_keys(
-            section_ids=section_ids
-        )
-        for section_id, list_item_id in section_keys_to_add:
+        for section_id, list_item_id in self.questionnaire_store_updater.started_section_keys(section_ids=section_ids):
+            # Only add sections which are repeated sections for this list, or the section in which this list is collected
+            # Prevents list item progresses being added as dependants as these are captured by started_section_keys(section_ids=section_ids)
+            if section_id == self.current_location.section_id and list_item_id or section_id != self.current_location.section_id and not list_item_id:
```

Comment:
I think variables and parentheses would make this clearer.

Is `section_id != self.current_location.section_id and not list_item_id:` correct?. This is could be another section which dependent the list?. Section id doesn't match and there is no list id which mean it is a normal section that depends on the section, for example it uses the list length for a when rule.

Reply:
yep good catch, not sure where I was going with that